sockeye/utils.py

```diff
@@ -615,3 +615,53 @@ def all_gather_object(obj: T) -> List[T]:
     obj_list = [None] * torch.distributed.get_world_size()  # type: List[T]
     torch.distributed.all_gather_object(obj_list, obj)
     return obj_list
+
+
+def count_seq_len(sample: str, count_type: str = 'char', replace_tokens: List = []) -> int:
+    """
+    Count sequence length, after replacing word segmentation token/s.
+    """
+    if len(replace_tokens) >= 1:
+        for r in replace_tokens:
+            sample = sample.replace(r, '')
+    if count_type == 'char':
+        return len(sample.replace(' ', ''))
+    else:
+        return len(sample.split(' '))
```

Comment:
there is a Sockeye-wide constant called TOKEN_SEPARATOR that maps to ' ' which you could use here.